geode-serialization/src/main/java/org/apache/geode/internal/serialization/filter/SerializableObjectConfig.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.serialization.filter;
+
+/**
+ * Defines the configuration properties for serialization filtering in Geode.
+ */
+public interface SerializableObjectConfig {
+
+  default String getSerializableObjectFilterIfEnabled() {
+    return getValidateSerializableObjects() ? getSerializableObjectFilter() : null;
+  }
+
+  boolean getValidateSerializableObjects();
+
+  void setValidateSerializableObjects(boolean value);
+
+  String getSerializableObjectFilter();
```

Comment:
We also need a setter `setSerializableObjectFilter`.  I have not seen `serializable-object-filter` being set in the product or test.

Reply:
`DistributionConfig` defines all of the methods that geode-core needs to call including `setSerializableObjectFilter` and `setValidateSerializableObjects`. We have lots of tests that are setting `serializable-object-filter`. Take a look at the callers of `DistributionConfig.setSerializableObjectFilter`. 

`SerializableObjectConfig` defines only the methods of `DistributionConfig` that geode-serialization needs to call and it does not currently need to invoke `setSerializableObjectFilter`. The next PR invokes `setValidateSerializableObjects` from geode-serialization but not `setSerializableObjectFilter`.

Reply:
Maybe I am missing something. No usage is found for `DistributionConfig.setSerializableObjectFilter` or `DistributionConfigImpl.setSerializableObjectFilter`. It is true that there are lots of test that are setting `serializable-object-filter`. But somehow those tests bypass the API `DistributionConfig.setSerializableObjectFilter` by doing a `Properties.put`. I am OK that we address the concern later though.